packages/jest/src/index.ts

```diff
@@ -17,3 +25,46 @@ export {
 export {run} from 'jest-cli';
 
 export type Config = ConfigTypes.InitialOptions;
+
+declare global {
+  export const beforeEach: typeof import('@jest/globals')['beforeEach'];
+  export const beforeAll: typeof import('@jest/globals')['beforeAll'];
+
+  export const afterEach: typeof import('@jest/globals')['afterEach'];
+  export const afterAll: typeof import('@jest/globals')['afterAll'];
+
+  export const describe: typeof import('@jest/globals')['describe'];
+  export const fdescribe: typeof import('@jest/globals')['fdescribe'];
+  export const xdescribe: typeof import('@jest/globals')['xdescribe'];
+
+  export const test: typeof import('@jest/globals')['test'];
+  export const xtest: typeof import('@jest/globals')['xtest'];
+
+  export const it: typeof import('@jest/globals')['it'];
+  export const fit: typeof import('@jest/globals')['fit'];
+  export const xit: typeof import('@jest/globals')['xit'];
+
+  export const expect: typeof import('@jest/globals')['expect'];
+
+  export const jest: typeof import('@jest/globals')['jest'];
+
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace jest {
```

Comment:
Not happy about the duplication with declarations in `@jest/globals`.

Another problem – it was not a good idea to put everything into `index` file. Now `import {Config} from 'jest'` is also augmenting `global` and pulling in Jest’s globals. That’s not good. So `"types": ["jest"]` should be reworked into something like `"types": ["jest/globals"]`. This means that `"exports"` needs additional entry.

In a way all is fine, but at the same time the implementation becomes more and more hairy. In contrary `import {expect, test} from '@jest/globals'` is simple and declarative. Does not need all the hairy stuff I mentioned. The more I work on this PR, the more I am a fan of that straight forward `import` ;D

Reply:
I think requiring `types: ['jest/global-api']` or something very explicit is perfectly fine 👍